docs/create-a-business-component.md

```diff
@@ -0,0 +1,345 @@
+
+### Making it dynamic
+
+Now that we have an Enhancer ready, we are going to use it in our StoreLocator.
+The major change here is the use of the data from the available props.
+
+When dealing with asynchronous resources like fetching data from the backend,
+you have to handle the loading state. Here we just show a simple `Loading..` to
+the user, but in a real life application, you would want to show placeholders or
+spinners.
+
+> You should also handle error cases, but it is not the purpose of this guide.
+> Just know that you could use
+> [Error Boundaries](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html)
+> for that.
+
+```js
+// src/theme/modules/StoreLocator/StoreLocator.js;
+
+import React from "react";
+import PropTypes from "prop-types";
+import { Map, TileLayer, Marker, Popup } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet/dist/images/marker-icon.png";
+import "leaflet/dist/images/marker-shadow.png";
+import StoreLocatoreQuery from "./StoreLocatorQuery.gql";
+import EnhanceStoreLocator from "./EnhanceStoreLocator";
+
+const StoreLocator = props => {
+  if (props.loading) {
+    return <div>Loading...</div>;
+  }
+
+  const coordinates = [
+    props.store.coordinates.longitude,
+    props.store.coordinates.latitude
+  ];
+  const defaultZoom = 14;
+
+  return (
+    <div>
+      <Map
+        center={coordinates}
+        zoom={defaultZoom}
+        style={{ height: "600px", width: "800px" }}
+      >
+        <TileLayer
+          attribution="&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        />
+        <Marker position={coordinates}>
+          <Popup>
+            <div>
+              My awesome store ─ {props.store.name}
+              Email: {props.store.owner.email}
+              Phone: {props.store.phone}
+            </div>
+          </Popup>
+        </Marker>
+      </Map>
+    </div>
+  );
+};
+
+StoreLocator.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  store: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    phone: PropTypes.string.isRequired,
+    owner: PropTypes.shape({
+      email: PropTypes.string.isRequired
+    }).isRequired,
+    coordinates: PropTypes.shape({
+      longitude: PropTypes.number.isRequired,
+      latitude: PropTypes.number.isRequired
+    }).isRequired
+  })
+};
+
+export default EnhanceStoreLocator(StoreLocatoreQuery)(StoreLocator);
+```
+
+> A side note on **Protypes**. The Proptypes allow us to validate the props
+> passed to a child component, this will avoid breaking your app passing invalid
+> props.  
+This will also serve as a props documentation for the other developers
+> in your team. They may be a little verbose, but it is important to maintain
+> them updated.
+
+## Using it in our App
+
+We now have all the required pieces, to finish, we just have to import our brand
```

Comment:
We now have extracted all the Store Locator logic. Now, we only have to use our brand and shiny module within the homepage.